core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1153,154 +1101,207 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
     finalLeaderIsrAndControllerEpoch
   }
 
-  class SessionExpirationListener() extends IZkStateListener with Logging {
-    this.logIdent = "[SessionExpirationListener on " + config.brokerId + "], "
+  private def checkAndTriggerPartitionRebalance(): Unit = {
```

Comment:
The reason I suggested this is that one of the commits added a missing `contains` check in this code. But the right fix is to use `get` instead of `apply`. The changes here are mostly about formatting the code correctly and avoiding bad APIs. That's pretty safe and it seems that we are just adding overhead by postponing it.